### PR TITLE
Glb morph targets

### LINKED ISSUE #179: Morph Target Support
We should add morph target support to the engine. We'll need POSITION0..7 semantics (preumably 8 is enough?), GLSL to do the interp. Obviously, we want the FBX conversion tool to read this data and write it to model JSON, but since the engine needs to work with procedurally created meshes, it's not a blocker.

### PR DESCRIPTION
**Pull request by slimbuck**
Fixes #179

The following changes are included in this PR:
- fixed morph target loading in the GLB viewer
- moved generateWireframe() function to the mesh
- updated the glb-viewer with new features and look

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Comment by willeastcott** [comment]
Does this also fix https://github.com/playcanvas/engine/issues/1045?

**Comment by slimbuck** [comment]
Most likely, but untested. Will do so now.

File: tools/glb-viewer/src/debug.js [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    mesh.vertexBuffer = new pc.VertexBuffer(device, vertexFormat, 1024, pc.BUFFER_DYNAMIC);
+    mesh.primitive[0].type = pc.PRIMITIVE_LINES;
+    mesh.primitive[0].base = 0;
+    mesh.primitive[0].indexed = false;
+    mesh.primitive[0].count = 0;
+
+    // construct the material
+    var material = new pc.BasicMaterial();
+    material.blend = false;
+    material.blendType = pc.BLEND_NORMAL;
```
Comment (willeastcott) [inline tools/glb-viewer/src/debug.js:48]: On the previous line, you disable blending. Then, setting `blendType` to `pc.BLEND_NORMAL` will reenable blending. Did you want `pc.BLEND_NONE`? https://github.com/playcanvas/engine/blob/master/src/scene/materials/material.js#L178

File: src/scene/mesh.js [hunk 2]
```diff
@@ -630,6 +629,65 @@ Object.assign(pc, function () {
...
+                    case pc.INDEXFORMAT_UINT32:
+                        return new Uint32Array(indexBuffer.storage);
+                    default:
+                        return null;
+                }
+            };
+
+            var lines = [];
+            var format;
+            if (this.indexBuffer[0]) {
```
Comment (mvaligursky) [inline src/scene/mesh.js:650]: maybe should be if (this.indexBuffer && this.indexBuffer[0]) to handle case when indexBuffer in undefined

**Comment by slimbuck** [comment]
Sorry I jumped the gun on this PR, will makes the suggested changes.

File: src/scene/mesh.js (hunk 2 above)
Comment (slimbuck) [reply src/scene/mesh.js:650]: this.indexBuffer is initialized to [null] at construction and (far as I know) should always be an array.

Adding the test could be safer, but it also implies to the reader that indexBuffer could validly be undefined or null. Is that the case?

Comment (slimbuck) [reply src/scene/mesh.js:650]: According to the doc the array could be null, so perhaps the test should be:
```this.indexBuffer.length > 0 && indexBuffer[0]``` ?

Comment (mvaligursky) [reply src/scene/mesh.js:650]: it's public variable, so the user could set it to just null.
I don't remember specific test case, but I had this issue with my code where indexBuffer was null, so it's safer to test it.

Comment (slimbuck) [reply src/scene/mesh.js:650]: I would argue that's a user error. What if a user sets indexBuffer to a string or map or anything else? We can't test all variables all the time and I think the code should reflect what the system considers to be valid state.

Comment (slimbuck) [reply src/scene/mesh.js:650]: And the valid state is defined by the property doc:
```
     * @property {pc.IndexBuffer[]} indexBuffer An array of index buffers. For unindexed meshes, this array can
     * be empty. The first index buffer in the array is used by {@link pc.MeshInstance}s with a renderStyle
     * property set to pc.RENDERSTYLE_SOLID. The second index buffer in the array is used if renderStyle is
     * set to pc.RENDERSTYLE_WIREFRAME.
```

Comment (mvaligursky) [reply src/scene/mesh.js:650]: fair enough.

Comment (slimbuck) [reply src/scene/mesh.js:650]: I actually think adding a check for null would do a disservice, because then a reader of the code would think that null is a valid state.

Comment (slimbuck) [reply src/scene/mesh.js:650]: I guess this is where typescript would help us; the setter function would have a signature disallowing any user from setting indexbuffer to null.

File: tools/glb-viewer/src/debug.js (hunk 1 above)
Comment (slimbuck) [reply tools/glb-viewer/src/debug.js:48]: this is being addressed in https://github.com/playcanvas/engine/pull/2024

File: src/scene/mesh.js (hunk 2 above)
Comment (slimbuck) [reply src/scene/mesh.js:650]: Additional array check is being added in https://github.com/playcanvas/engine/pull/2024
